packages/lit-virtualizer/src/lib/repeat.ts

```diff
@@ -0,0 +1,147 @@
+import { directive, NodePart, createMarker, TemplateResult } from 'lit-html';
+import { VirtualRepeater } from './uni-virtualizer/lib/VirtualRepeater';
+import { Layout } from './uni-virtualizer/lib/layouts/Layout';
+
+/**
+ * Mixin for VirtualRepeater and VirtualScroller. This mixin overrides the generic
+ * methods in those classes to provide lit-specific implementations of element
+ * creation and manipulation.
+ * 
+ * This mixin implements child recycling, so children can be reused after removal
+ * from the DOM.
+ */
+export const LitMixin = Superclass => class extends Superclass {
+  // NodeParts that are available for reuse.
+  _pool: Array<NodePart>;
+
+  // Method for generating each item's DOM.
+  _template: (item: any, index?: number) => TemplateResult;
+
+  // Children are rendered into this part.
+  // The host of the directive that constructs the scroller.
+  _hostPart: NodePart;
+  
+  /**
+   * TODO @straversi: This is not a true mixin. Mixins do have information
+   * about potential superclasses. This class appears to know the constructor
+   * arguments of an unknown superclass. It also overrides methods on an
+   * unknown superclass.
+   */
+  constructor(config: {
+    part: NodePart,
+    template: (item: any, index?: number) => TemplateResult,
+    useShadowDOM?: boolean,
+    scrollTarget?: Element | Window,
+    layout?: Layout
+  }) {
+    const {part, template, useShadowDOM, layout} = config;
+    let container = part.startNode.parentNode;
+    let scrollTarget = config.scrollTarget || container;
+    super({container, scrollTarget, useShadowDOM, layout});
+
+    this._pool = [];
+    this._template = template;
+    this._hostPart = part;
+  }
+
+  createElement(): NodePart {
+    return this._pool.pop() || new NodePart(this._hostPart.options);
+  }
+
+  updateElement(part: NodePart, item, idx: number) {
+    part.setValue(this._template(item, idx));
+    part.commit();
+  }
+
+  recycleElement(part: NodePart) {
+    this._pool.push(part);
+  }
+
+  /*
+   * Lit-specific overrides for node manipulation
+   */
+
+  get _kids(): Array<Node> {
```

Comment:
More descriptive name? I immediately think anything named `kids` should be named `children`, but this isn't. How's it different from the concept of `children`?

Reply:
My recollection is fuzzy, but at the time this abstraction was introduced I think there was a terminology clash and (being in prototyping mode) it seemed more expedient to name this something silly but distinct than to do the necessary cleanup work.

In any case, I think other things have since been renamed and this can indeed be changed to `children` (here and in the base implementation).